### ISSUE TITLE
NVDA reports the text as a heading but is unable to navigate to it when h is pressed

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#5186]
**Reported by pranavlal on 2015-06-27 06:25**
I am attaching a word document to this ticket. When I move the cursor over the text, nvda says heading 1. However, if I use quick navigation, I am told that there are no headings in the document. This happens across versions of Microsoft word namely 2010 and 2013.

**Comment by michaelDCurran** [comment]
So I don't forget (again)... Technical:
* It seems impossible to detect the built-in style used for a range of text in Microsoft Word. You can get to the style object, but that does not expose any property reflecting the built-in style constant. There is a nameLocal property, but this is localized for the current language.

Some further notes:
* range.goto (wdGotoHeading) only locates ranges of text directly using one of the heading built-in style constants (I.e. heading1 through heading9). It does not detect styles based off these heading styles.
* MS Word 2016's UIA implementation only exposes a UIA heading style for the heading built-in styles, again, not ones that are based off of these styles.
 
If it were possible to detect the built-in style used for a range of text, we decided that we would only expose heading levels in NVDA for MS Word if the range used one of the heading built-in style constants directly, so as to match goto (used by heading quicknav).

**Comment by nvaccessAuto** [comment]
Incubated in 8352d381474d756a52b9ce87d3d5e39d7cc75051.

**Comment by michaelDCurran** [comment]
In the end we did end up going with matching on the localized name of builtin heading styles, but by creating a cache of builtin heading style localized names on first use.
